### PR TITLE
Authentication and Users Controllers

### PR DESCRIPTION
**Pull request by shrestaz**
**Note**: database data script has been updated to create Users table with Username, password and salt. (previously it was name and password). So make you your users table is updated otherwise the request **WILL** fail. (I simply recreated the entire database)

----

This PR introduces authentication controller and users controllers.
You can create users by making a POST request at endpoint `/api/auth/users` with JSON body
```json
{
    "username": "SomeUsername",
    "password": "SomePassword"
}
```

image

you can get users in 3 ways:
- all users: `/api/users`
- user by id: `/api/users/userId/{UserId}`
- user by username: `/api/users/username/{Username}`

### CODE REVIEW

File: Subproject2/3. Data Layer/Database Context/SOVAContext.cs [hunk 1]
```diff
@@ -116,6 +115,9 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
             modelBuilder.Entity<Submission>().HasOne(a => a.SoMember).WithMany().HasForeignKey(a => a.SoMemberId);
             modelBuilder.Entity<UserHistory>().HasOne(a => a.User).WithMany().HasForeignKey(a => a.UserId);
             modelBuilder.Entity<UserHistory>().HasOne(a => a.History).WithMany().HasForeignKey(a => a.HistoryId);
+
+            modelBuilder.Entity<User>().Property(u => u.Username).HasColumnName("name");
```
Comment (ozgey99) [inline Subproject2/3. Data Layer/Database Context/SOVAContext.cs]: you can remove this line now

File: Subproject2/1. SOVA/Controllers/CommentsController.cs [hunk 2]
```diff
@@ -15,7 +15,7 @@ namespace _1._SOVA.Controllers
     public class CommentsController : ControllerBase
     {
         private readonly ICommentRepository _commentRepository;
-        private IMapper _mapper;
+        private readonly IMapper _mapper;
```
Comment (ozgey99) [inline Subproject2/1. SOVA/Controllers/CommentsController.cs]: why do we need readonly here?

Comment (shrestaz) [reply Subproject2/1. SOVA/Controllers/CommentsController.cs]: Resharper changed that to readonly. It doesn't really affect the functionality but I'll change it back.

File: Subproject2/3. Data Layer/Database Context/SOVAContext.cs (hunk 1 above)
Comment (shrestaz) [reply Subproject2/3. Data Layer/Database Context/SOVAContext.cs]: Done on my local changes. Will push when finished with clean up

**Review by ozgey99 — APPROVED** [review]

**Pull request merged**